internal/helpers/httpclient.go

```diff
@@ -0,0 +1,59 @@
+package helpers
+
+import (
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/gokrazy/internal/config"
+	"github.com/gokrazy/internal/httpclient"
+	"github.com/gokrazy/internal/updateflag"
+)
+
+func GetHTTPClient(instance string) (*http.Client, *url.URL, error) {
+	_, updateHostname := updateflag.GetUpdateTarget(instance)
+	const configBaseName = "http-password.txt"
+	pw, err := config.HostnameSpecific(updateHostname).ReadFile(configBaseName)
```

Comment:
nit: just use "http-password.txt" directly (without the constant) for consistency